sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -132,180 +160,72 @@ public boolean isEndRequired() {
 
   @Override
   public void shutdown() {
-    workerThread.interrupt();
-    worker.shutdown();
+    forceFlush();
+    ScheduledFuture<?> nextExport = this.nextExport;
+    if (nextExport != null) {
+      nextExport.cancel(false);
+    }
+    executor.shutdown();
+    spanExporter.shutdown();
   }
 
   @Override
   public void forceFlush() {
-    worker.forceFlush();
+    exportBatch();
   }
 
-  // Worker is a thread that batches multiple spans and calls the registered SpanExporter to export
-  // the data.
-  //
-  // The list of batched data is protected by an explicit monitor object which ensures full
-  // concurrency.
-  private static final class Worker implements Runnable {
-
-    static {
-      Meter meter = OpenTelemetry.getMeter("io.opentelemetry.sdk.trace");
-      LongCounter droppedSpansCounter =
-          meter
-              .longCounterBuilder("droppedSpans")
-              .setUnit("1")
-              .setDescription(
-                  "The number of spans dropped by the BatchSpanProcessor due to high throughput.")
-              .build();
-      droppedSpans =
-          droppedSpansCounter.bind(
-              Labels.of("spanProcessorType", BatchSpanProcessor.class.getSimpleName()));
-    }
-
-    private static final BoundLongCounter droppedSpans;
-
-    private final Timer timer = new Timer(TIMER_THREAD_NAME, /* isDaemon= */ true);
-
-    private static final Logger logger = Logger.getLogger(Worker.class.getName());
-    private final SpanExporter spanExporter;
-    private final long scheduleDelayMillis;
-    private final int maxQueueSize;
-    private final int maxExportBatchSize;
-    private final int halfMaxQueueSize;
-    private final Object monitor = new Object();
-    private final int exporterTimeoutMillis;
-    private final AtomicBoolean exportAvailable = new AtomicBoolean(true);
-
-    @GuardedBy("monitor")
-    private final List<ReadableSpan> spansList;
-
-    private Worker(
-        SpanExporter spanExporter,
-        long scheduleDelayMillis,
-        int maxQueueSize,
-        int maxExportBatchSize,
-        int exporterTimeoutMillis) {
-      this.spanExporter = spanExporter;
-      this.scheduleDelayMillis = scheduleDelayMillis;
-      this.maxQueueSize = maxQueueSize;
-      this.halfMaxQueueSize = maxQueueSize >> 1;
-      this.maxExportBatchSize = maxExportBatchSize;
-      this.exporterTimeoutMillis = exporterTimeoutMillis;
-      this.spansList = new ArrayList<>(maxQueueSize);
+  // Timeout is best effort don't need to handle the future return value.
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private void exportBatch() {
+    List<ReadableSpan> batch = new ArrayList<>(Math.min(maxExportBatchSize, queue.size()));
+    if (queue.drainTo(batch, maxExportBatchSize) == 0) {
```

Comment:
Should benchmark this, but we can save on allocations if we take elements by one, convert them to `SpanData` and only then put to outgoing `forExport`. Then we don't need `batch`.